src/common/transformations/src/transformations/common_optimizations/reshape_sequence_fusion.cpp

```diff
@@ -17,21 +17,53 @@
 NGRAPH_RTTI_DEFINITION(ngraph::pass::ReshapeSequenceFusion, "ReshapeSequenceFusion", 0);
 
 namespace {
+OPENVINO_SUPPRESS_DEPRECATED_START
 bool has_valid_pattern(const std::shared_ptr<ngraph::Node> & node) {
     auto const_node = std::dynamic_pointer_cast<ngraph::opset8::Constant>(node);
-    if (!const_node) return false;
+    if (!const_node) {
+        // Lower bound of the value
+        ngraph::HostTensorPtr lb = std::make_shared<ngraph::HostTensor>(node->output(0));
+        const bool is_lb_evaluated = node->evaluate_lower({lb});
+        if (!is_lb_evaluated || !lb) return false;
+        const auto lb_const_node = std::make_shared<ngraph::opset8::Constant>(lb);
+        if (!lb_const_node) return false;
+
+        const auto & lb_values = lb_const_node->cast_vector<int64_t>();
+        bool is_valid = std::all_of(lb_values.cbegin(), lb_values.cend(), [](int64_t value) { return value > 0;});
```

Comment:
Please add comments about this constraints.

Reply:
Sure, done